### PR TITLE
Migrates to null safety

### PR DESCRIPTION
**Pull request by Ascenio**

### CODE REVIEW

**Comment by Ascenio** [comment]
@googlebot I signed it!

File: lib/network.dart [hunk 1]
```diff
@@ -71,76 +74,78 @@ class NetworkImageWithRetry extends ImageProvider<NetworkImageWithRetry> {
 
   @override
   ImageStreamCompleter load(NetworkImageWithRetry key, DecoderCallback decode) {
-    return OneFrameImageStreamCompleter(
-        _loadWithRetry(key, decode),
+    final Future<ImageInfo> maybeImage =
+        _loadWithRetry(key, decode).then((ImageInfo? value) => value!);
+    return OneFrameImageStreamCompleter(maybeImage,
```
[the remark this answers is not in the document]
Comment (escamoteur) [reply lib/network.dart:79]: What's the state here?

[the remark this answers is not in the document]
Comment (escamoteur) [reply lib/network.dart:79]: Is it possible that value can ever be null? Of so the null '!' would be fatal

[the remark this answers is not in the document]
Comment (escamoteur) [reply lib/network.dart:79]: After discussion with @slightfoot I resolve this here

**Comment by escamoteur** [comment]
I merged this to a separate branch so that it can be accessed already via a github reference. @mit-mit it to me it looks fine so I would recommend merging it into master and pushing a new version.

**Comment by mit-mit** [comment]
@Ascenio can you open a PR to take this to master, please?

**Comment by Ascenio** [comment]
@mit-mit so, just make the same PR again?

**Comment by mit-mit** [comment]
Yes, but against the master channel. That is the one we release to pub.dev from. Now that null safety has launched, no reason to keep this in a special branch.

**Pull request merged**